index.html

```diff
@@ -0,0 +1,578 @@
+
+  1. rely on a hardcoded interpretation for directives with certain signatures, or
+  2. accept additional configuration about how to interpret directives in the schema.
+
+The first solution is fragile, particularly as GraphQL has no built-in namespacing mechanisms, so the possibility of name collisions always looms.
+
+The second is unfortunate: GraphQL schemas are generally intended to be self-describing, and requiring additional configuration subtly undermines this guarantee: given just a schema, programs do not necessarily know how to interpret it, and certainly not how to serve it. It also creates the possibility for the schema and configuration to fall out of sync, leading to issues which can manifest late in a deployment pipeline.
+
+## Core Schemas
+
+Introducing **core schemas**.
+
+<div class=hbox>
+  <a class=core>
+    <div class=ring></div>
+    <div class=nucleus>core schema</div>
+  </a>
+</div>
+
+A basic core schema:
+
+<view-of src=basic.graphql title='Basic core schema'></view-of>
+
+**Core schemas** provide a clear and concise mechanism for schema documents to specify the intended meaning of the directives which they contain. Additionally, they provide:
+  - [**Flexible namespacing rules.**](#prefixing) It is always possible to represent any GraphQL schema within a core schema document. Additionally, documents can [choose the names](#as) they use for the directives they reference, guaranteeing that namespace collisions can always be resolved.
+  - [**Directive versioning.**](#versioning) Directive specifications follow [semver-like semantic versioning principles](#versioning), which helps schema processors determine if they are able to correctly interpret a document's metadata.
+  - [**Standard rules for extensibility**](#extensibility) of directives and their associated input types. These rules allow specs to annotate each others' metadata, effectively providing "directives on directives," which are otherwise impossible within the GraphQL schema language
+
+**Core schemas are not a new language.** All core schema documents are valid GraphQL schema documents. However, this specification introduces new requirements, so not all valid GraphQL schemas are valid core schemas.
+
+The broad intention behind core schemas is to provide a *single document* which provides all the necessary configuration of a *data core*—some program which serves the schema to GraphQL clients, primarily by following directives in order to determine how to resolve queries made against that schema.
+
+### Parts of a Core Schema
+
+When talking about a core schema, we can broadly break it into two pieces:
```

Comment:
So right now, you're defining the core schema as the union of two disjoint sets: the export schema and the machinery". (Or at least that's how I'm interpreting this section.)

I think this currently doesn't work that great given that non-exported fields already makes the description as written false. They are (output) fields that are not in the export schema, not directives or input types. (Are they machinery? I'm not even sure.)

And I think this dichotomy will be even less helpful when we move on to a more blueprint/constellation world where fields are getting renamed and transformed etc.

So I think instead of saying that the core schema is these two things put together, we should say that the export schema is something that can be produced from the core schema based on rules defined by the specs that are used. Those rules are typically "remove all my directives" but it could be something else. For example, this spec says that anything annotated with `@core(exports: false)` is removed, and that all the `@core` directives themselves are removed.  (But you could imagine a schema that allows for some sort of rename directive that says you rename things rather than remove them in order to produce the export schema.)

And this should be in a section somewhere! I don't think it says anywhere "all `@core` directives are removed from the export schema". I don't think this should be implicit — specs should be allowed to define things that stay in the export schema.

This line of thinking actually makes me think that the "not in export schema" directive should *not* be part of `@core` itself.  If each schema has its own rules about "things to do as part of building the export schema if you support this spec", then "things with a particular directive are definitely dropped" feels pretty separate to me.  So we can have a separate `exported` spec with one directive, and that directive would say "the rules that this spec adds to the 'make the export schema' process is, you drop all the things annotated with my directive".  (This also solves my other concern where I don't like `@core` being used for both concepts :) )

Reply:
As discussed in a comment thread on https://apollographql.quip.com/6jvoAVhBie45/CSDL-Core-Naming (sorry, Apollo internal) the idea that export should be a simple subset of core based on a single specified directive (and that any fancier features should be implemented in the generator of the core schema rather than the router) works for me, so I mostly withdraw this feedback, though I do still feel like this spec should offer a bit more focus/motivation on the idea that defining an export schema is a key part of what core schemas do.

(And I do still think the directive needs a different name from the other one.)